scripts/migrate_maas_instance_ids.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+
+#This script script is intended to run only against controller machines that
+#have upgraded from a MAAS 1.9 provider to a MAAS 2.x provider. Do not use this
+#script otherwise. The javascript below is horrible. It is designed to run
+#operations in bulk targeting a minimum version of mongo 3.2 which has little
+#functionality in the way of aggregation.
+
+while getopts 'r' flag; do
+    case "${flag}" in
+        r) rollback_flag=true
+	   ;;
+        *) echo "This script is dangerous use it carefully" $>2
+	   ;;
+    esac
+done
+
+if [ $rollback_flag ]
+then
+    #This script warps a MAAS 1.9 style URL around the instance_id. You can
+    #run this script with the -r flag for convenience but you should really just reload the
+    #backup that you have created with mongoexport or otherwise if you run into trouble.
+    mongo_script='db.instanceData.aggregate([{"$project":{instanceid:{"$cond":{if:{"$ne":[{"$substr":["$instanceid",0,20]},"/MAAS/api/1.0/nodes/"]},then:{"$concat":["/MAAS/api/1.0/nodes/","$instanceid","/"]},else:"$instanceid"}},document:"$$ROOT"}},{"$out":"tmp"}]);ops=db.instanceData.find().map(function(item){newid=db.tmp.findOne({"document.instanceid":item.instanceid});return{"updateOne":{"filter":{instanceid:newid.document.instanceid},"update":{"$set":{instanceid:newid.instanceid}}}}});db.instanceData.bulkWrite(ops);'
+else
+    #This script drops the URL wrapping the instance which coerces the data into MAAS 2.0 format
+    mongo_script='db.instanceData.aggregate([{"$project":{instanceid:{"$cond":{if:{"$eq":[{"$substr":["$instanceid",0,20]},"/MAAS/api/1.0/nodes/"]},then:{"$substr":["$instanceid",20,-1]},else:"$instanceid"}},document:"$$ROOT"}},{"$out":"tmp"}]);ops=db.instanceData.find().map(function(item){newid=db.tmp.findOne({"document.instanceid":item.instanceid});if(item.instanceid.slice(-1)==="/"){return{"updateOne":{"filter":{instanceid:newid.document.instanceid},"update":{"$set":{instanceid:newid.instanceid.slice(0,-1)}}}}}else{return{}}});db.instanceData.bulkWrite(ops);'
```

Comment:
any chance we could change these to some sort of multi-line strings that would make it easier to read for humans?